x-pack/plugins/search_playground/public/components/view_query/view_query_flyout.tsx

```diff
@@ -68,14 +77,39 @@ export const ViewQueryFlyout: React.FC<ViewQueryFlyoutProps> = ({ onClose }) =>
       ...tempQueryFields,
       [index]: newFields,
     });
+    usageTracker.count('view_query_fields_updated', newFields.length);
   };
 
   const saveQuery = () => {
     queryFieldsOnChange(tempQueryFields);
     elasticsearchQueryChange(createQuery(tempQueryFields, fields));
     onClose();
+
+    usageTracker.click('view_query_save');
+    usageTracker.count(
+      'view_query_sparse_fields',
+      Object.values(tempQueryFields)
+        .flat()
+        .filter((field) => SUGGESTED_SPARSE_FIELDS.includes(field)).length
+    );
+    usageTracker.count(
+      'view_query_bm25_fields',
+      Object.values(tempQueryFields)
+        .flat()
+        .filter((field) => SUGGESTED_BM25_FIELDS.includes(field)).length
+    );
+    usageTracker.count(
+      'view_query_dense_vector_fields',
+      Object.values(tempQueryFields)
+        .flat()
+        .filter((field) => SUGGESTED_DENSE_VECTOR_FIELDS.includes(field)).length
```

Comment:
dont filter on these SUGGESTED arrays to determine if its bm25 / elser / sparse. Need to group based on the source_fields to determine whether its elser / dense / bm25